pyvista/plotting/render_window_interactor.py

```diff
@@ -305,26 +310,66 @@ def update_style(self):
-            interaction with all charts).
+        mouse_pos : tuple of float
+            Tuple containing the mouse position.
 
         """
-        scene = None
+        # Loop over all renderers to see whether any charts need to be made interactive
+        interactive_scene = None
         for renderer in self._plotter.renderers:
-            if scene is None and mouse_pos is not None and renderer.IsInViewport(*mouse_pos):
-                scene = renderer._charts.toggle_interaction(mouse_pos)
+            if interactive_scene is None and renderer.IsInViewport(*mouse_pos):
+                # No interactive charts yet and mouse is within this renderer's viewport,
+                # so collect all charts indicated by the mouse (typically only one, except
+                # when there are overlapping charts).
+                origin = renderer.GetOrigin()  # Correct for viewport origin (see #3278)
+                charts = renderer._get_charts_by_pos(
+                    (mouse_pos[0] - origin[0], mouse_pos[1] - origin[1])
+                )
+                if charts:
+                    # Toggle interaction for indicated charts and determine whether
+                    # there are any remaining interactive charts.
+                    interactive_charts = renderer.set_chart_interaction(charts, toggle=True)
+                    if interactive_charts:
+                        # Save a reference to this renderer's scene if there are
+                        # remaining interactive charts.
+                        interactive_scene = renderer._charts._scene
+                else:
+                    # No indicated charts, so disable interaction with all charts
+                    # for this renderer.
+                    renderer.set_chart_interaction(False)
             else:
-                # Not in viewport or already an active chart found (in case they overlap), so disable interaction
-                renderer._charts.toggle_interaction(False)
+                # Not in viewport or interactive charts were already found in another
+                # renderer, so disable interaction with all charts for this renderer.
+                renderer.set_chart_interaction(False)
+        # Manually set context_style based on found interactive scene (or stop interaction
+        # with any scene if there are no interactive charts).
+        self._set_context_style(interactive_scene)
+
+    def _set_context_style(self, scene):
+        """
+        Set the context style interactor or switch back to previous interactor style.
 
+        Parameters
+        ----------
+        scene : vtkContextScene, optional
+            The scene to interact with or ``None`` to stop interaction with any scene.
+
+        """
         # Set scene to interact with or reset it to stop interaction (otherwise crash)
+        if vtk_version_info < (9, 3, 0):
+            # TODO: determine in which VTK version !9811 will be available
```

Comment:
@banesullivan Do you know whether the next VTK version will be 9.2.6 or 9.3.0? Then I can update this check accordingly :)

Reply:
I don't think that was backported to the `release` branch, so I'd put my money on `9.3`